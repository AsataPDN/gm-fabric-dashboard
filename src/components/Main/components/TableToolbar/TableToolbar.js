// NPM Modules
import React from "react";
import { PropTypes } from "prop-types";

// Internal Components
import SearchInput from "./components/GMSearchInput";
import Button from "../../../Button";
import Form from "./components/Form";
import SecondaryText from "components/SecondaryText";
import GMSelect from "components/Main/components/GMSelect";
import Toolbar from "./components/Toolbar";
import ToolbarLeft from "./components/ToolbarLeft";
import ToolbarRight from "./components/ToolbarRight";
import ToolbarCenter from "./components/ToolbarCenter";

TableToolbar.propTypes = {
  displayType: PropTypes.oneOf(["Card", "Table"]), // if this is a Fabric Grid Toolbar, display type buttons will be rendered
  filterString: PropTypes.string, // value for controlled SearchInput input
  groupByAttribute: PropTypes.string, // value for controlled GMSelect Group By dropdown
  groupByOptions: PropTypes.array, // array of option objects for the GMSelect Group By dropdown
  groupByProps: PropTypes.object, // optional props that will be passed down to SearchInput component
  searchInputProps: PropTypes.object, // optional props that will be passed down to GMSelect Group By dropdown
  setDisplayType: PropTypes.func, // onClick handler for ToolbarCenter buttons. Takes in either “Card” or “Table”.
  setFilterString: PropTypes.func, // onChange handler for search input
  setGroupByAttribute: PropTypes.func, // onChange handler for GMSelect Group By dropdown
  setSortByAttribute: PropTypes.func, // onChange handler for GMSelect Sort By dropdown
  sortByAttribute: PropTypes.string, // value for controlled GMSelect Sort By dropdown
  sortByOptions: PropTypes.array, // array of option objects for the GMSelect Sort By dropdown
  sortByProps: PropTypes.object // optional props that will be passed down to GMSelect Sort By dropdown
};

export default function TableToolbar({
  displayType,
  setDisplayType,
  filterString,
  searchInputProps,
  setFilterString,
  groupByAttribute,
  groupByOptions,
  groupByProps,
  setGroupByAttribute,
  sortByAttribute,
  sortByOptions,
  sortByProps,
  setSortByAttribute
}) {
  return (
    <Toolbar>
      <ToolbarLeft>
        {setFilterString &&
          filterString != null && (
            <Form>
              <SearchInput
                className="form-control"
                {...searchInputProps}
                onChange={evt => setFilterString(evt.target.value)}
                placeholder="Search Services"
                aria-label="Search All Services"
                value={filterString}
              />
            </Form>
          )}
      </ToolbarLeft>
      {setDisplayType &&
        displayType && (
          <ToolbarCenter>
            <Button
              active={displayType === "Card"}
              clickAction={() => setDisplayType("Card")}
              glyph="Card"
              label="Card"
            />
            <Button
              active={displayType === "Table"}
              clickAction={() => setDisplayType("Table")}
              glyph="List"
              label="List"
            />
          </ToolbarCenter>
        )}
      <ToolbarRight>
        {setGroupByAttribute &&
          groupByOptions &&
          groupByAttribute && (
            <GMSelect
              {...groupByProps}
              name="form-field-group-by"
              options={groupByOptions}
              value={groupByAttribute}
              onChange={val => setGroupByAttribute(val.value)}
              clearable={false}
              searchable={false}
              valueRenderer={val => (
                <span>
                  <span>Group </span>
                  <SecondaryText>{val.label}</SecondaryText>
                </span>
              )}
            />
          )}
        {setSortByAttribute &&
          sortByOptions &&
          sortByAttribute && (
            <GMSelect
              {...sortByProps}
              name="form-field-sort-by"
              options={sortByOptions}
              value={sortByAttribute}
              onChange={val => setSortByAttribute(val.value)}
              clearable={false}
              searchable={false}
              valueRenderer={val => (
                <span>
                  <span>Sort </span>
                  <SecondaryText>{val.label}</SecondaryText>
                </span>
              )}
            />
          )}
      </ToolbarRight>
    </Toolbar>
  );
}
