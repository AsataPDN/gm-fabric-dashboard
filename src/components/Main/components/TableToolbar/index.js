// NPM Modules
import React from "react";
import { PropTypes } from "prop-types";

// Internal Components
import SearchInput from "./components/GMSearchInput";

// External Dependencies
import Button from "components/Button";
import Form from "./components/Form";
import SecondaryText from "components/SecondaryText";
import GMSelect from "../components/../GMSelect";
import Toolbar from "./components/Toolbar";
import ToolbarLeft from "./components/ToolbarLeft";
import ToolbarRight from "./components/ToolbarRight";
import ToolbarCenter from "./components/ToolbarCenter";

TableToolBar.propTypes = {
  displayType: PropTypes.oneOf(["Card", "Table"]), // if this is a Fabric Grid toolbar, display type buttons will be rendered
  filterString: PropTypes.string, // value for controlled SearchInput input
  groupByAttribute: PropTypes.string, // value for controlled GMSelect Group By dropdown
  groupByOptions: PropTypes.array, // array of option objects for the GMSelect Group By dropdown
  groupByProps: PropTypes.object, // optional props that will be passed down to SearchInput component
  searchInputProps: PropTypes.object, // optional props that will be passed down to GMSelect Group By dropdown
  setDisplayType: PropTypes.func, // onClick handler for ToolbarCenter buttons. Takes in either “Card” or “Table”.
  setFilterString: PropTypes.func, // onChange handler for search input
  setGroupByAttribute: PropTypes.func, // onChange handler for GMSelect Group By dropdown
  setSortByAttribute: PropTypes.func, // onChange handler for GMSelect Sort By dropdown
  sortByAttribute: PropTypes.string, // value for controlled GMSelect Sort By dropdown
  sortByOptions: PropTypes.array, // array of option objects for the GMSelect Sort By dropdown
  sortByProps: PropTypes.object // optional props that will be passed down to GMSelect Sort By dropdown
};

TableToolBar.defaultProps = {
  displayType: "",
  filterString: ""
};

export default function TableToolBar({
  displayType,
  filterString,
  setDisplayType,
  groupByAttribute,
  setFilterString,
  setGroupByAttribute,
  setSortByAttribute,
  sortByAttribute,
  statusView
}) {
  return (
    <Toolbar>
      <ToolbarLeft>
        <Form>
          <SearchInput
            className="form-control"
            onChange={evt => setFilterString(evt.target.value)}
            placeholder="Search Services"
            aria-label="Search All Services"
            value={filterString}
          />
        </Form>
      </ToolbarLeft>
      <ToolbarCenter>
        <Button
          active={displayType === "Card"}
          clickAction={() => setDisplayType("Card")}
          glyph="Card"
          label="Card"
        />
        <Button
          active={displayType === "Table"}
          clickAction={() => setDisplayType("Table")}
          glyph="List"
          label="List"
        />
      </ToolbarCenter>
      <ToolbarRight>
        <GMSelect
          name="form-field-group-by"
          options={
            statusView
              ? [
                  {
                    value: "Owner",
                    label: "Owner"
                  },
                  {
                    value: "Status",
                    label: "Status"
                  },
                  {
                    value: "Capability",
                    label: "Capability"
                  }
                ]
              : [
                  {
                    value: "Owner",
                    label: "Owner"
                  },
                  {
                    value: "Capability",
                    label: "Capability"
                  },
                  {
                    value: "Status",
                    label: "Status"
                  },
                  {
                    value: "None",
                    label: "None"
                  }
                ]
          }
          value={groupByAttribute}
          onChange={val => setGroupByAttribute(val.value)}
          clearable={false}
          searchable={false}
          valueRenderer={val => (
            <span>
              <span>Group </span>
              <SecondaryText>{val.label}</SecondaryText>
            </span>
          )}
        />
        <GMSelect
          name="form-field-sort-by"
          options={[
            {
              value: "Name",
              label: "Name"
            },
            {
              value: "Status",
              label: "Status"
            }
          ]}
          value={sortByAttribute}
          onChange={val => setSortByAttribute(val.value)}
          clearable={false}
          searchable={false}
          valueRenderer={val => (
            <span>
              <span>Sort </span>
              <SecondaryText>{val.label}</SecondaryText>
            </span>
          )}
        />
        )}
      </ToolbarRight>
    </Toolbar>
  );
}
