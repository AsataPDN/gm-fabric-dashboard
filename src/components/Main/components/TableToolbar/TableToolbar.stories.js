import React from "react";

import { storiesOf } from "@storybook/react";

import TableToolbar from "./index.js";

import { withKnobs, text, object, select } from "@storybook/addon-knobs";

const sortByOptions = [
  {
    value: "Name",
    label: "Name"
  },
  {
    value: "Status",
    label: "Status"
  }
];
const groupByOptions = [
  {
    value: "Owner",
    label: "Owner"
  },
  {
    value: "Capability",
    label: "Capability"
  },
  {
    value: "Status",
    label: "Status"
  },
  {
    value: "None",
    label: "None"
  }
];
storiesOf("Table Toolbar", module)
  .addDecorator(withKnobs)
  .add("Fabric Main View", () => {
    return (
      <div style={{ width: "100%" }}>
        <TableToolbar
          displayType={select("displayType", ["Card", "List"], "Card")}
          setDisplayType={() => alert("Fired setDisplayType()")}
          setFilterString={() => alert("Fired setFilterString()")}
          filterString={text("filterString", "string to search")}
          groupByOptions={object("groupByOptions", groupByOptions)}
          groupByAttribute={select(
            "groupByAttribute",
            ["Owner", "Capability", "Status", "None"],
            "Status"
          )}
          setGroupByAttribute={() => alert("Fired setGroupByAttribute()")}
          sortByOptions={object("sortByOptions", sortByOptions)}
          sortByAttribute={select(
            "sortByAttribute",
            ["Name", "Status"],
            "Status"
          )}
          setSortByAttribute={() => alert("Fired setSortByAttribute()")}
        />
      </div>
    );
  })
  .add("Fabric Status View", () => {
    return (
      <div style={{ width: "100%" }}>
        <TableToolbar
          displayType={select("displayType", ["Card", "List"], "Card")}
          setDisplayType={() => alert("Fired setDisplayType()")}
          setFilterString={() => alert("Fired setFilterString()")}
          filterString={text("filterString", "string to search")}
          groupByOptions={object("groupByOptions", groupByOptions)}
          groupByAttribute={select(
            "groupByAttribute",
            ["Owner", "Capability", "Status", "None"],
            "Status"
          )}
          setGroupByAttribute={() => alert("Fired setGroupByAttribute()")}
        />
      </div>
    );
  })
  .add("Service/Function/Routes View", () => {
    return (
      <div style={{ width: "100%" }}>
        <TableToolbar
          setDisplayType={() => alert("Fired setDisplayType()")}
          setFilterString={() => alert("Fired setFilterString()")}
          filterString={text("filterString", "string to search")}
          sortByOptions={object("sortByOptions", sortByOptions)}
          sortByAttribute={select(
            "sortByAttribute",
            ["Name", "Status"],
            "Status"
          )}
          setSortByAttribute={() => alert("Fired setSortByAttribute()")}
        />
      </div>
    );
  });
