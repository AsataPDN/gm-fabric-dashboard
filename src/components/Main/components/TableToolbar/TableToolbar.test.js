import React from "react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";

import TableToolbar from "./index.js";

const options = [
  {
    value: "SampleVal1",
    label: "SampleVal1"
  },
  {
    value: "SampleVal2",
    label: "SampleVal2"
  }
];

const mockFabricViewProps = {
  displayType: "Card",
  setDisplayType: jest.fn(),
  filterString: "",
  searchPlaceholder: "Search",
  setFilterString: jest.fn(),
  groupByAttribute: "SampleVal1",
  groupByOptions: options,
  setGroupByAttribute: jest.fn(),
  sortByAttribute: "SampleVal2",
  sortByOptions: options,
  setSortByAttribute: jest.fn()
};

describe("Table Toolbar", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(<TableToolbar {...mockFabricViewProps} />);
  });

  test("matches snapshot", () => {
    const tree = renderer
      .create(<TableToolbar {...mockFabricViewProps} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders search box", () => {
    expect(wrapper.find({ type: "search" })).toHaveLength(1);
  });

  test("does not render search box if setFilterString, searchPlaceholder, or filterString are not provided", () => {
    wrapper.setProps({
      setFilterString: null,
      filterString: null,
      searchPlaceholder: null
    });
    expect(wrapper.find({ type: "search" })).toHaveLength(0);
  });

  test("renders display type buttons", () => {
    // Find buttons with title attributes "Card" and "List"
    expect(wrapper.find("button").find({ title: "Card" })).toHaveLength(1);
    expect(wrapper.find("button").find({ title: "List" })).toHaveLength(1);
  });

  test("does not render display type buttons if setDisplayType or displayType are not provided", () => {
    wrapper.setProps({
      setDisplayType: null,
      displayType: null
    });
    // Find buttons with title attributes "Card" and "List"
    expect(wrapper.find("button").find({ title: "Card" })).toHaveLength(0);
    expect(wrapper.find("button").find({ title: "List" })).toHaveLength(0);
  });

  test("adds an active class to button that matches displayType", () => {
    let cardButton = wrapper.find("button").find({ title: "Card" });
    let listButton = wrapper.find("button").find({ title: "List" });
    expect(cardButton.hasClass("active")).toBe(true);
    expect(listButton.hasClass("active")).toBe(false);
    wrapper.setProps({ displayType: "List" });
    listButton = wrapper.find("button").find({ title: "List" });
    cardButton = wrapper.find("button").find({ title: "Card" });
    expect(listButton.hasClass("active")).toBe(true);
    expect(cardButton.hasClass("active")).toBe(false);
  });

  test("renders a group by dropdown with a value equal to groupByAttribute", () => {
    const dropdown = wrapper
      .find("Select")
      .find("input")
      .find({ name: "form-field-group-by" });
    expect(dropdown).toHaveLength(1);
    expect(dropdown.props().value).toBe("SampleVal1");
  });

  test("does not render a group by dropdown if setGroupByAttribute, groupByOptions, or groupByAttribute is not provided", () => {
    wrapper.setProps({
      setGroupByAttribute: null,
      groupByOptions: null,
      groupByAttribute: null
    });
    const dropdown = wrapper
      .find("Select")
      .find("input")
      .find({ name: "form-field-group-by" });
    expect(dropdown).toHaveLength(0);
  });

  test("renders a sort by dropdown", () => {
    const dropdown = wrapper
      .find("Select")
      .find("input")
      .find({ name: "form-field-sort-by" });
    expect(dropdown).toHaveLength(1);
    expect(dropdown.props().value).toBe("SampleVal2");
  });

  test("does not render a group by dropdown if setSortByAttribute, sortByOptions, or sortByAttribute is not provided", () => {
    wrapper.setProps({
      setSortByAttribute: null,
      sortByOptions: null,
      sortByAttribute: null
    });
    const dropdown = wrapper
      .find("Select")
      .find("input")
      .find({ name: "form-field-sort-by" });
    expect(dropdown).toHaveLength(0);
  });
});
