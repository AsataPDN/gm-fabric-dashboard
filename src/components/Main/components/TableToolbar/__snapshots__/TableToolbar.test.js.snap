// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Toolbar matches snapshot 1`] = `
<div
  className="sc-EHOje dMHSiZ"
>
  <div
    className="sc-bZQynM VzWwq"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        aria-label="Search Services"
        className="sc-bdVaJa dexvml"
        onChange={[Function]}
        placeholder="Search Services"
        type="search"
        value=""
      />
    </form>
  </div>
  <div
    className="sc-htoDjs dxGVYy"
  >
    <button
      className="btn active"
      disabled={undefined}
      onClick={[Function]}
      style={undefined}
      tabIndex={undefined}
      title="Card"
    >
      <svg
        aria-labelledby="ariaLabelledby"
        className="sc-htpNat FamBZ"
        focusable="false"
        version="1.1"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
        xmlnsXlink="http://www.w3.org/1999/xlink"
      >
        <g
          className="sc-bwzfXH blkLhm"
          fill="currentColor"
          title=""
        >
          <title>
            
          </title>
          <g
            className="glyph"
            fill="currentColor"
            transform="translate(0 0) scale(1)"
          >
            <path
              d="M7,4 L17,4 C18.1045695,4 19,4.8954305 19,6 L19,17 L19,17 C19,18.1045695 18.1045695,19 17,19 L7,19 L7,19 C5.8954305,19 5,18.1045695 5,17 L5,6 L5,6 C5,4.8954305 5.8954305,4 7,4 L7,4 Z M10.5,16 L16.5,16 C16.7761424,16 17,16.2238576 17,16.5 C17,16.7761424 16.7761424,17 16.5,17 L10.5,17 C10.2238576,17 10,16.7761424 10,16.5 C10,16.2238576 10.2238576,16 10.5,16 Z M7.5,10 L14.5,10 C14.7761424,10 15,10.2238576 15,10.5 C15,10.7761424 14.7761424,11 14.5,11 L7.5,11 C7.22385763,11 7,10.7761424 7,10.5 C7,10.2238576 7.22385763,10 7.5,10 Z M7.5,12 L10.5,12 C10.7761424,12 11,12.2238576 11,12.5 C11,12.7761424 10.7761424,13 10.5,13 L7.5,13 C7.22385763,13 7,12.7761424 7,12.5 C7,12.2238576 7.22385763,12 7.5,12 Z M8,6 L16,6 C16.5522847,6 17,6.44771525 17,7 L17,8 C17,8.55228475 16.5522847,9 16,9 L8,9 C7.44771525,9 7,8.55228475 7,8 L7,7 C7,6.44771525 7.44771525,6 8,6 Z M7,5 L7,5 C6.44771525,5 6,5.44771525 6,6 L6,17 C6,17.5522847 6.44771525,18 7,18 L17,18 C17.5522847,18 18,17.5522847 18,17 L18,6 C18,5.44771525 17.5522847,5 17,5 L7,5 Z"
              id="path-1"
            />
          </g>
        </g>
      </svg>
      <span
        className="label"
      >
        
        Card
        
      </span>
    </button>
    <button
      className="btn"
      disabled={undefined}
      onClick={[Function]}
      style={undefined}
      tabIndex={undefined}
      title="List"
    >
      <svg
        aria-labelledby="ariaLabelledby"
        className="sc-htpNat FamBZ"
        focusable="false"
        version="1.1"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
        xmlnsXlink="http://www.w3.org/1999/xlink"
      >
        <g
          className="sc-bwzfXH blkLhm"
          fill="currentColor"
          title=""
        >
          <title>
            
          </title>
          <g
            className="glyph"
            fill="currentColor"
            transform="translate(0 0) scale(1)"
          >
            <path
              d="M5.5,8 C4.67157288,8 4,7.32842712 4,6.5 C4,5.67157288 4.67157288,5 5.5,5 C6.32842712,5 7,5.67157288 7,6.5 C7,7.32842712 6.32842712,8 5.5,8 Z M5.5,13 C4.67157288,13 4,12.3284271 4,11.5 C4,10.6715729 4.67157288,10 5.5,10 C6.32842712,10 7,10.6715729 7,11.5 C7,12.3284271 6.32842712,13 5.5,13 Z M5.5,18 C4.67157288,18 4,17.3284271 4,16.5 C4,15.6715729 4.67157288,15 5.5,15 C6.32842712,15 7,15.6715729 7,16.5 C7,17.3284271 6.32842712,18 5.5,18 Z M9,5.5 L19,5.5 C19.5522847,5.5 20,5.94771525 20,6.5 L20,6.5 C20,7.05228475 19.5522847,7.5 19,7.5 L9,7.5 C8.44771525,7.5 8,7.05228475 8,6.5 L8,6.5 L8,6.5 C8,5.94771525 8.44771525,5.5 9,5.5 Z M9,10.5 L19,10.5 C19.5522847,10.5 20,10.9477153 20,11.5 C20,12.0522847 19.5522847,12.5 19,12.5 L9,12.5 C8.44771525,12.5 8,12.0522847 8,11.5 C8,10.9477153 8.44771525,10.5 9,10.5 Z M9,15.5 L19,15.5 C19.5522847,15.5 20,15.9477153 20,16.5 C20,17.0522847 19.5522847,17.5 19,17.5 L9,17.5 C8.44771525,17.5 8,17.0522847 8,16.5 C8,15.9477153 8.44771525,15.5 9,15.5 Z"
              id="Combined-Shape"
            />
          </g>
        </g>
      </svg>
      <span
        className="label"
      >
        
        List
        
      </span>
    </button>
  </div>
  <div
    className="sc-gzVnrw cElssY"
  >
    <div
      className="Select sc-bxivhb iHWWTX Select--single has-value"
      style={undefined}
    >
      <input
        disabled={false}
        name="form-field-group-by"
        type="hidden"
        value="Status"
      />
      <div
        className="Select-control"
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={undefined}
      >
        <span
          className="Select-multi-value-wrapper"
          id="react-select-4--value"
        >
          <div
            className="Select-value"
            style={undefined}
            title={undefined}
          >
            <span
              aria-selected="true"
              className="Select-value-label"
              id="react-select-4--value-item"
              role="option"
            >
              <span>
                <span>
                  Group
                   
                </span>
                <span
                  className="sc-ifAKCX dAlnhc"
                >
                  Status
                </span>
              </span>
            </span>
          </div>
          <div
            aria-activedescendant="react-select-4--value"
            aria-disabled="false"
            aria-expanded={false}
            aria-label={undefined}
            aria-labelledby={undefined}
            aria-owns=""
            className="Select-input"
            onBlur={[Function]}
            onFocus={[Function]}
            role="combobox"
            style={
              Object {
                "border": 0,
                "display": "inline-block",
                "width": 1,
              }
            }
            tabIndex={0}
          />
        </span>
        <span
          className="Select-arrow-zone"
          onMouseDown={[Function]}
        >
          <span
            className="Select-arrow"
            onMouseDown={[Function]}
          />
        </span>
      </div>
    </div>
    <div
      className="Select sc-bxivhb iHWWTX Select--single has-value"
      style={undefined}
    >
      <input
        disabled={false}
        name="form-field-sort-by"
        type="hidden"
        value="Name"
      />
      <div
        className="Select-control"
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={undefined}
      >
        <span
          className="Select-multi-value-wrapper"
          id="react-select-5--value"
        >
          <div
            className="Select-value"
            style={undefined}
            title={undefined}
          >
            <span
              aria-selected="true"
              className="Select-value-label"
              id="react-select-5--value-item"
              role="option"
            >
              <span>
                <span>
                  Sort
                   
                </span>
                <span
                  className="sc-ifAKCX dAlnhc"
                >
                  Name
                </span>
              </span>
            </span>
          </div>
          <div
            aria-activedescendant="react-select-5--value"
            aria-disabled="false"
            aria-expanded={false}
            aria-label={undefined}
            aria-labelledby={undefined}
            aria-owns=""
            className="Select-input"
            onBlur={[Function]}
            onFocus={[Function]}
            role="combobox"
            style={
              Object {
                "border": 0,
                "display": "inline-block",
                "width": 1,
              }
            }
            tabIndex={0}
          />
        </span>
        <span
          className="Select-arrow-zone"
          onMouseDown={[Function]}
        >
          <span
            className="Select-arrow"
            onMouseDown={[Function]}
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;
